r_markdown/background_analysis_ecoc.Rmd

```diff
@@ -0,0 +1,273 @@
+
+## Convert to ecoc to longer format
+
+```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
+  names(ecoc_merged)[names(ecoc_merged) == 'date'] <- 'analysis_date'
+  ecoc_merged <- tidyr::gather(ecoc_merged, "pol", "val", one_of("ec_ug_sq_cm","oc_ug_sq_cm"))
+```
+
+# Extract all ecoc tests
+
+```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}  
+  ec <- filter(ecoc_merged, grepl("^[0-9]|^G", id) == TRUE, pol == "ec_ug_sq_cm")
+  oc <- filter(ecoc_merged, grepl("^[0-9]|^G", id) == TRUE, pol == "oc_ug_sq_cm")
+```
+
+# Filter out test ec artifact filters and correct oc filters for artifact
+
+```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}  
+  ec <- filter(ec, cassette == "E")    
+  oc_e <- filter(oc, cassette == "E")
+  oc_a <- filter(oc, cassette == "A")
+  oc <- dplyr::left_join(oc_e, oc_a, by = "id")
+  oc <- filter(oc, !is.na(cassette.y))
+  oc$delta_oc <- oc$val.x - oc$val.y
+```
+
+# Merge and calculate
+
+```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
+  ec <- left_join(ec, select(samples, id, fuel, stove, date), by = "id")
+  oc <- left_join(oc, select(samples, id, fuel, stove, date), by = "id")
+  ec <- left_join(ec, select(data_1, id, time_start_cart_white_orange, time_end_cart_white_orange), by = "id")
+  oc <- left_join(oc, select(data_1, id, time_start_cart_white_orange, time_end_cart_white_orange), by = "id")
+  names(ec)[names(ec) == 'time_start_cart_white_orange'] <- 'sample_start'
+  names(oc)[names(oc) == 'time_start_cart_white_orange'] <- 'sample_start'
+  names(ec)[names(ec) == 'time_end_cart_white_orange'] <- 'sample_end'
+  names(oc)[names(oc) == 'time_end_cart_white_orange'] <- 'sample_end'
+```
+
+```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
+  ec$test_length <- ec$sample_end - ec$sample_start
+  ec$ec_rate <- ec$val / (ec$test_length /(60*60))
+  oc$test_length <- oc$sample_end - oc$sample_start
+  oc$oc_rate <- oc$delta_oc / (oc$test_length /(60*60))
+```
+
+# Extract background tests and ecoc data
+
+```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}  
+  ec_bg <- filter(ec, grepl("^G", id) == TRUE, pol == "ec_ug_sq_cm")
+  oc_bg <- filter(oc, grepl("^G", id) == TRUE, pol.x == "oc_ug_sq_cm")
+```
+
+
+## Plot background ecoc rate (units in ug per sq cm per hour)
+
+```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}  
+  ggplot(ec_bg, aes(x=ec_rate)) +
+    geom_histogram(alpha = 0.4) +
+    theme_minimal() + ggtitle("EC rate")
+
+  ggplot(oc_bg, aes(x=oc_rate)) +
+    geom_histogram(alpha = 0.4) +
+    theme_minimal() + ggtitle("OC rate")
+```
+
+## Plot background ecoc levels by date
+
+```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}  
+  ggplot(ec_bg, aes(date, ec_rate, color = qc)) +
+    geom_point() +
+    theme_minimal()
+  ggplot(oc_bg, aes(date, oc_rate, color = qc.x)) +
```

Comment:
I think it usually just adds .x .y if you are joining and columns have the same name but i can fix